codegen.ml

```diff
@@ -300,8 +300,17 @@ let translate (program) =
         (match op with 
           A.Add     -> L.build_fadd
         | A.Sub     -> L.build_fsub
-        | A.Mul    -> L.build_fmul
+        | A.Mul     -> L.build_fmul
         | A.Div     -> L.build_fdiv 
+        | A.Mod     -> L.build_frem
+        | A.Eq      -> L.build_fcmp L.Fcmp.Oeq
+        | A.Neq     -> L.build_fcmp L.Fcmp.One
+        | A.Less    -> L.build_fcmp L.Fcmp.Olt
+        | A.Leq     -> L.build_fcmp L.Fcmp.Ole
+        | A.Greater -> L.build_fcmp L.Fcmp.Ogt
+        | A.Geq     -> L.build_fcmp L.Fcmp.Oge
+        | A.And | A.Or ->
+            raise (Failure "internal error: semant should have rejected and/or on float")
```

Comment:
Should we reject everything expect boolean?

Reply:
I think the other group (former project) did like this and not sure if we need to clarify that.